src/tasks/minions/minigames/infernoActivity.ts

```diff
@@ -105,28 +107,32 @@ export const infernoTask: MinionTask = {
 			}
 		}
 
-		if (isOnTask) {
-			const points = await calculateSlayerPoints(user.user.slayer_last_task, usersTask.slayerMaster!, user);
+		if (isOnTask && !deathTime) {
 			const { newUser } = await user.update({
-				slayer_points: {
-					increment: points
-				},
 				slayer_task_streak: {
 					increment: 1
 				}
 			});
 
+			const currentStreak = newUser.slayer_task_streak;
+			const points = await calculateSlayerPoints(currentStreak, usersTask.slayerMaster!, user);
+			const secondNewUser = await user.update({
+				slayer_points: {
+					increment: points
+				}
+			});
+
 			await prisma.slayerTask.update({
 				where: {
 					id: usersTask.currentTask!.id
 				},
 				data: {
 					quantity_remaining: 0,
-					skipped: deathTime ? true : false
+					skipped: false
 				}
 			});
 
-			text += `\n\n**You've completed ${newUser.slayer_task_streak} tasks and received ${points} points; giving you a total of ${newUser.slayer_points}; return to a Slayer master.**`;
+			text += `\n\n**You've completed ${currentStreak} tasks and received ${points} points; giving you a total of ${secondNewUser.newUser.slayer_points}; return to a Slayer master.**`;
```

Comment:
```suggestion
			text += `\n\n**You've completed ${currentStreak} tasks and received ${points} points; giving you a total of ${secondNewUser.slayer_points}; return to a Slayer master.**`;
```
This doesn't look right, same in fightCavesActivity

Reply:
The const `newUser` is already deconstructed from the object returned from updating the slayer task streaks. `secondNewUser` is the actually object returned from updating the points which contains the newUser property which is why I have to call it here. This is the same way that `monsterActivity.tsx` does it.